plugins/inputs/win_perf_counters/win_perf_counters.go

```diff
@@ -132,6 +133,34 @@ func (m *Win_PerfCounters) SampleConfig() string {
 	return sampleConfig
 }
 
+func expandCounterQuery(query string) ([]string, error) {
+	var bufSize uint32
+	var buf []uint16
+	ret := PdhExpandWildCardPath(query, nil, &bufSize)
+	for ret == PDH_MORE_DATA {
+		buf = make([]uint16, bufSize)
+		ret = PdhExpandWildCardPath(query, &buf[0], &bufSize)
+	}
+	if ret == ERROR_SUCCESS {
+		return UTF16ToStringArray(buf), nil
+	}
+	return nil, fmt.Errorf("Failed to expand query: '%s', err(%d)", query, ret)
+}
+
+func formatCounterQuery(objectname string, instance string, counter string) string {
+	if instance == "------" {
+		return "\\" + objectname + "\\" + counter
+	}
+	return "\\" + objectname + "(" + instance + ")\\" + counter
+}
+
+func expandQuery(query string, expand bool) ([]string, error) {
```

Comment:
Hi it looks like this could be the solution for #2879 the only thing would be to check if the instance name equals "*" and than expand the query. The expandQuery and formatCounterQuery Method could handle this together.